spec/builtin-fc.test.sh

```diff
@@ -0,0 +1,258 @@
+## tags: interactive
+## compare_shells: bash
+## oils_failures_allowed: 2
+
+#### fc -l
+rm -f tmp
+
+echo '
+HISTFILE=tmp
+seq 3 > tmp
+history -c
+history -r
+
+fc -l
+' | $SH --norc -i
+
+# match osh's behaviour of echoing ^D for EOF
+case $SH in bash) echo '^D' ;; esac
+
+## STDOUT:
+1	 history -r
+2	 1
+3	 2
+4	 3
+^D
+## END
+
+#### fc -ln
+rm -f tmp
+
+echo '
+HISTFILE=tmp
+seq 3 > tmp
+history -c
+history -r
+
+fc -ln
+' | $SH --norc -i
+
+# match osh's behaviour of echoing ^D for EOF
+case $SH in bash) echo '^D' ;; esac
+
+## STDOUT:
+	 history -r
+	 1
+	 2
+	 3
+^D
+## END
+
+#### fc -lr
+rm -f tmp
+
+echo '
+HISTFILE=tmp
+seq 3 > tmp
+history -c
+history -r
+
+fc -lr
+' | $SH --norc -i
+
+# match osh's behaviour of echoing ^D for EOF
+case $SH in bash) echo '^D' ;; esac
+
+## STDOUT:
+4	 3
+3	 2
+2	 1
+1	 history -r
+^D
+## END
+
+#### fc -lnr
+rm -f tmp
+
+echo '
+HISTFILE=tmp
+seq 3 > tmp
+history -c
+history -r
+
+fc -lnr
+' | $SH --norc -i
+
+# match osh's behaviour of echoing ^D for EOF
+case $SH in bash) echo '^D' ;; esac
+
+## STDOUT:
+	 3
+	 2
+	 1
+	 history -r
+^D
+## END
+
+#### fc -l with default offset
```

Comment:
"fc  -l without args uses the default command history offset" (if that is accurate)